src/services/shopify-client.ts

```diff
@@ -21,3 +24,45 @@ export type ShopifyErrorResult = { cannotConnectToAPI: true };
 export interface ShopifyClient {
   getShopifyData(): Promise<Result<ConnectedShop[], ShopifyErrorResult>>;
 }
+
+export class HttpShopifyClient implements ShopifyClient {
+  private readonly axios: AxiosInstance;
+  private readonly baseUrl: string;
+  private readonly connectionDataRef: RefObject<AppSession>;
+
+  constructor({
+    axiosStatic,
+    baseUrl,
+    connectionDataRef,
+  }: {
+    axiosStatic: AxiosStatic;
+    baseUrl: string;
+    connectionDataRef: RefObject<AppSession>;
+  }) {
+    this.baseUrl = baseUrl;
+    this.axios = axiosStatic.create({
+      baseURL: this.baseUrl,
+    });
+    this.connectionDataRef = connectionDataRef;
+  }
+
+  private getShopifyConnectionData(): ShopifyConnectionData {
+    const connectionData = this.connectionDataRef.current;
+    if (!connectionData || connectionData.status !== 'authenticated' || !connectionData.jwtToken) {
+      throw new Error('Shopify connection data is not available');
+    }
+    return connectionData;
+  }
+
+  public async getShopifyData(): Promise<Result<ConnectedShop[], ShopifyErrorResult>> {
+    const { jwtToken } = this.getShopifyConnectionData();
+    const response = await this.axios.request<Result<ConnectedShop[], ShopifyErrorResult>>({
```

Comment:
I think that the type is wrong, is the response is successful you have to map it to a `ConnectedShop[]`, if it is not (or if there is an exception), you have to map it to a `ShopifyErrorResult`

Reply:
Added happy path. Pending error.